lib/datadog/appsec/configuration.rb

```diff
@@ -79,6 +79,10 @@ def configure
           settings
         end
 
+        def default_setting?(setting)
+          settings.default?(setting)
+        end
```

Comment:
I was expecting the API do be more like this:

```
Datadog.configuration.appsec.enabled # => false
Datadog.configuration.appsec.default?(:enabled) # => true

Datadog.configure do |c|
  c.appsec.enabled = false
end

Datadog.configuration.appsec.enabled # => false
Datadog.configuration.appsec.default?(:enabled) # => true
```

And, ultimately, working across the board for all DD settings, and then public.

The way it is now is fine for our current purposes, but I'd expect it to be internal. Making it `private` though is a bit of a problem because it's going to be called within a variety of internal places, where we'd need to use `send(:default_setting?, :foo)` all over the place.

Reply:
@lloeki @ivoanjo 

On my initial attempt, I had it as above:

```
Datadog.configuration.appsec.enabled # => false
Datadog.configuration.appsec.default?(:enabled) # => true

Datadog.configure do |c|
  c.appsec.enabled = false
end

Datadog.configuration.appsec.enabled # => false
Datadog.configuration.appsec.default?(:enabled) # => true
```

I saw that it wasn't consistent with the rest of the `Datadog.configuration` components. I started looking at a way to make it available to all components. The biggest blocker was that AppSec defines the settings differently than the rest of the components, so it would require more changes. 

So I first thought of making it "private" by encapsulating it in the AppSec module.  

Later we can figure out how to make the API identical across the different components.

Reply:
There are indeed a few places in the codebase where we're using `send(...)` exactly because we're marking stuff as private.

I think in this case given that a) We're not very sure we want this exact shape for things and b) we're talking about `Datadog::AppSec`, I think `private`-and-`send` solution is what I recommend we go with.